src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -75,9 +78,14 @@ func parseDuration(r *http.Request, key string) (time.Duration, error) {
 		return value, nil
 	}
 
-	// Try parsing as an integer value specifying seconds, the Prometheus default
-	if seconds, intErr := strconv.ParseInt(str, 10, 64); intErr == nil {
-		return time.Duration(seconds) * time.Second, nil
+	// Try parsing as a float value specifying seconds, the Prometheus default
+	if seconds, floatErr := strconv.ParseFloat(str, 64); floatErr == nil {
```

Comment:
I think there's a bug here, you're not returning `floatErr` here if it's not nil

also, nit: when you have a decent size codeblock I think it might read a little nicer to flip the conditional and then split this up, i.e.

```golang
seconds, err := strconv.ParseFloat(str, 64)
if err != nil {
 return nil, err
}
// logic
```

Reply:
Eh i mean it only gets to here if this returns an error:
```
	value, err := time.ParseDuration(str)
```
So either way it'll return an error - just might not be the float conversion error.

Reply:
That's wrong. It needs to return this error